sdk/appconfiguration/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationReplicaClientsBuilder.java

```diff
@@ -206,29 +216,76 @@ public void setEnvironment(Environment environment) {
                 break;
             }
         }
+        this.env = environment;
     }
 
-    @Override
     protected ConfigurationClientBuilder createBuilderInstance() {
-        ConfigurationClientBuilder builder = new ConfigurationClientBuilder();
+        RetryStrategy retryStatagy = null;
+
+        String mode = env.getProperty(AzureGlobalProperties.PREFIX + "." + RETRY_MODE_PROPERTY_NAME);
+        String modeService = env.getProperty(AzureAppConfigurationProperties.PREFIX + "." + RETRY_MODE_PROPERTY_NAME);
+
+        if ("exponential".equals(mode) || "exponential".equals(modeService) || (mode == null && modeService == null)) {
+            int retries = checkPropertyInt(MAX_RETRIES_PROPERTY_NAME, defaultMaxRetries);
+
+            Duration baseDelay = checkPropertyDuration(BASE_DELAY_PROPERTY_NAME, DEFAULT_MIN_RETRY_POLICY);
+            Duration maxDelay = checkPropertyDuration(MAX_DELAY_PROPERTY_NAME, DEFAULT_MAX_RETRY_POLICY);
 
-        ExponentialBackoff retryPolicy = new ExponentialBackoff(maxRetries, DEFAULT_MIN_RETRY_POLICY,
-            DEFAULT_MAX_RETRY_POLICY);
+            retryStatagy = new ExponentialBackoff(retries, baseDelay, maxDelay);
+        }
+
+        ConfigurationClientBuilder builder = clientFactory.build();
 
-        builder.retryPolicy(new RetryPolicy(retryPolicy));
+        if (retryStatagy != null) {
+            builder.retryPolicy(new RetryPolicy(retryStatagy));
+        }
 
         return builder;
     }
 
-    @Override
-    protected void configureHttpLogOptions(ConfigurationClientBuilder builder) {
-        ClientOptionsProvider.ClientOptions client = getAzureProperties().getClient();
-
-        if (client instanceof ClientOptionsProvider.HttpClientOptions) {
-            HttpLogOptions logOptions = HTTP_LOG_OPTIONS_CONVERTER
-                .convert(((ClientOptionsProvider.HttpClientOptions) client).getLogging());
-            consumeHttpLogOptions().accept(builder, logOptions);
+    private int checkPropertyInt(String propertyName, int defaultValue) {
+        String envValue = env.getProperty(AzureGlobalProperties.PREFIX + "." + propertyName);
+        String envServiceValue = env.getProperty(AzureAppConfigurationProperties.PREFIX + "." + propertyName);
+        int value = defaultValue;
+
+        if (envServiceValue != null) {
+            try {
+                value = Integer.parseInt(envServiceValue);
+            } catch (NumberFormatException e) {
+                LOGGER.warn("{}.{} isn't a valid integer, using default value.", AzureAppConfigurationProperties.PREFIX,
+                    propertyName);
+            }
+        } else if (envValue != null) {
+            try {
+                value = Integer.parseInt(envValue);
+            } catch (NumberFormatException e) {
+                LOGGER.warn("{}.{} isn't a valid integer, using default value.", AzureGlobalProperties.PREFIX,
+                    propertyName);
+            }
         }
+        return value;
```

Comment:
These two seem like follow the same pattern, should we extract one general method?